crates/api_models/src/payment_methods.rs

```diff
@@ -207,6 +207,18 @@ pub struct ResponsePaymentMethodTypes {
     pub bank_debits: Option<BankDebitTypes>,
     /// The Bank transfer payment method information, if applicable for a payment method type.
     pub bank_transfers: Option<BankTransferTypes>,
+
+    /// Required fields for the payment_method_type.
+    pub required_fields: Option<Vec<RequiredFieldInfo>>,
+}
+
+// Required fields info used while listing the payment methods
+#[derive(Debug, serde::Deserialize, serde::Serialize, Clone, PartialEq, Eq)]
+pub struct RequiredFieldInfo {
+    pub required_field: Option<String>,
+    pub display_name: Option<String>,
+    pub field_type: Option<String>,
+    pub field_options: Option<Vec<String>>,
```

Comment:
This should be an enum

instead of 
```
field_type : Option<String>,
field_options: Option<Vec<String>>
```

It could be
```
enum Field {
    TEXT,
    CHOICE{options: Vec<String>}
}
```

also do all the fields here need to be optional? seem like they should be mandatory

Reply:
field_type could be enum right? like

field_type: FieldType
enum FieldType{
    Text,
    Dropdown
}

and field_options will remain same. is this how u r suggesting?